src/test/java/atlastests/CohortDefinitionStepDefs.java

```diff
@@ -346,23 +346,38 @@ public void clickToGenerateImpalaButton() {
         $(By.xpath("//*[@class='cohort-generate-sources']/tbody/tr/td/span/span/button")).click();
     }
 
+    @When("^click to Generate first data source button$")
+    public void clickToGenerateFirstDataSourceButton() {
+        $(By.xpath("//*[@class='cohort-generate-sources']/tbody/tr/td/span/span/button")).click();
+    }
+
     @Then("^can see Complete in IMPALA status in (\\d+) seconds$")
     public void canSeeCompleteStatusInSeconds(int arg0) {
         $(By.xpath("//*[@class='cohort-generate-sources']/tbody/tr/td[3]")).waitUntil(text("COMPLETE"), arg0 * 1000);
     }
 
+    @Then("^can see Complete in first data source status in (\\d+) seconds$")
+    public void canSeeCompleteFirstDSStatusInSeconds(int arg0) {
+        $(By.xpath("//*[@class='cohort-generate-sources']/tbody/tr/td[3]")).waitUntil(text("COMPLETE"), arg0 * 1000);
```

Comment:
please, if it's possible - rewrite every new method without a "hardcode" like this "td/tr[2]"